src/internal/packager/git/fetch.go

```diff
@@ -22,7 +22,30 @@ func (g *Git) fetchRef(ref string) error {
 		return g.fetchHash(ref)
 	}
 
-	return g.fetchTag(ref)
+	err := g.fetchTag(ref)
+	if err == nil {
+		return nil
+	}
+
+	return g.fetchBranch(ref)
+}
```

Comment:
Instead of doing error fall-through here, I would prefer modifying `isHash` to be smarter and to use it in the parent of this function to determine ref type (then make this a switch/case statement instead).  This should be possible if we figure out what kind of ref we have prior to `g.removeLocalTagRefs()` in the parent caller of this function. (i.e. head/tag/hash) https://stackoverflow.com/questions/18222634/given-a-git-refname-can-i-detect-whether-its-a-hash-tag-or-branch

Reply:
(let me know if you would like clarification and I can mock up an example of what this would look like)

Reply:
Can add that in this PR or happy to make an issue to cleanup how that's done in a separate issue. Agree there's room for improvement